packages/gen-ai-hub/src/utils/model.ts

```diff
@@ -0,0 +1,37 @@
+import { AiDeployment } from '@sap-ai-sdk/ai-core';
+
+/**
+ * A foundation model is identified by its name and potentially a version.
+ */
+export interface FoundationModel {
+  /**
+   * The name of the model.
+   */
+  name: string;
+  /**
+   * The version of the model.
+   */
+  version?: string;
+}
+
+function isFoundationModel(
+  model: Partial<FoundationModel> | undefined
+): model is FoundationModel {
+  return typeof model === 'object' && 'name' in model;
+}
+
+/**
+ * Get the model information from a deployment.
+ * @param deployment - AI core model deployment.
+ * @returns The model information.
+ * @internal
+ */
+export function extractModel(
+  // TODO: this type does not seem to be correct (backend_details => backendDetails), check why
```

Comment:
I checked and it seems to be an actual mistake in the API spec. Requested a fix.